packages/rainbowkit/src/wallets/useWalletConnectors.ts

```diff
@@ -1,37 +1,55 @@
 import { Connector, useConnect } from 'wagmi';
-import { flatten } from '../utils/flatten';
 import { indexBy } from '../utils/indexBy';
-import { isNotNullish } from '../utils/isNotNullish';
 import {
   useInitialChainId,
   useRainbowKitChains,
 } from './../components/RainbowKitProvider/RainbowKitChainContext';
-import { WalletInstance } from './Wallet';
+import { WagmiConnectorInstance, WalletInstance } from './Wallet';
 import {
   getDesktopDownloadUrl,
   getExtensionDownloadUrl,
   getMobileDownloadUrl,
 } from './downloadUrls';
+import {
+  connectorsWithRecentWallets,
+  isEIP6963Connector,
+  isRainbowKitConnector,
+  isRecentWallet,
+  rainbowKitConnectorWithWalletConnect,
+} from './groupedWallets';
 import { addRecentWalletId, getRecentWalletIds } from './recentWalletIds';
 
 export interface WalletConnector extends WalletInstance {
   ready?: boolean;
-  connect?: ReturnType<typeof useConnect>['connectAsync'];
-  onConnecting?: (fn: () => void) => void;
+  connect: () => Promise<{
+    accounts: readonly `0x${string}`[];
+    chainId: number;
+  }>;
   showWalletConnectModal?: () => void;
   recent: boolean;
   mobileDownloadUrl?: string;
   extensionDownloadUrl?: string;
   desktopDownloadUrl?: string;
+  getDesktopUri?: () => Promise<string>;
+  getQrCodeUri?: () => Promise<string>;
+  getMobileUri?: () => Promise<string>;
 }
 
+const eip6963StrPrefix = 'eip6963';
+
 export function useWalletConnectors(): WalletConnector[] {
   const rainbowKitChains = useRainbowKitChains();
   const intialChainId = useInitialChainId();
   const { connectAsync, connectors: defaultConnectors_untyped } = useConnect();
-  const defaultConnectors = defaultConnectors_untyped as Connector[];
+  const defaultCreatedConnectors =
```

Comment:
Some refactoring regarding to [this](https://github.com/rainbow-me/rainbowkit/pull/1601/files#r1394090065)